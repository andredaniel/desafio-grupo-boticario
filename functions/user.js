const axios = require('axios')

const randomStatus = () => {
  const key = Math.floor(Math.random() * 3)

  const statuses = ['APPROVED', 'DISAPPROVED', 'IN_VALIDATION']

  return statuses[key]
}

exports.handler = async function (event, context) {
  const baseUrl = 'https://5fffb468cb21e10017af7e99.mockapi.io/api'

  if (event.httpMethod === 'GET') {
    const { data } = await axios.get(
      `${baseUrl}/orders/?user_id=${event.queryStringParameters.user_id}`
    )

    return {
      statusCode: 200,
      body: JSON.stringify(data),
    }
  }

  if (event.httpMethod === 'POST') {
    const newOrder = JSON.parse(event.body)

    newOrder.cashbackAppliedPercentage = Math.ceil(Math.random() * 20)
    newOrder.cashbackReceivedValue =
      (newOrder.value / 100) * newOrder.cashbackAppliedPercentage
    newOrder.status = randomStatus()

    const { data } = await axios.post(`${baseUrl}/orders`, newOrder)

    return {
      statusCode: 200,
      body: JSON.stringify(data),
    }
  }
}
