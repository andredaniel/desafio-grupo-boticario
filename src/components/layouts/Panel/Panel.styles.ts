import styled from 'styled-components'

export const PanelWrapper = styled.div`
  padding: 80px 0 30px;

  @media (min-width: 768px) {
    padding: 80px 50px 80px 300px;
  }
`

export const AppContent = styled.div`
  background-color: #fff;
  border-radius: 25px;
  min-height: calc(100vh - 170px);
  padding: 30px 20px;
  position: relative;

  @media (min-width: 768px) {
    min-height: auto;
    padding: 50px;
  }

  h1.title {
    margin-top: 0;
  }
`
