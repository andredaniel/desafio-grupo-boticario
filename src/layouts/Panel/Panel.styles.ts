import styled from 'styled-components'
import { theme } from '../../theme'

export const PanelWrapper = styled.div`
  padding: 80px 0 30px;

  @media (min-width: ${theme.breakpoints.md}) {
    padding: 80px 50px 80px 300px;
  }
`

export const AppContent = styled.div<{ overflow?: string }>`
  background-color: #fff;
  border-radius: 25px;
  height: calc(100vh - 170px);
  max-height: calc((var(--vh, 1vh) * 100) - 170px);
  overflow-y: ${({ overflow }) => (overflow ? overflow : 'scroll')};
  padding: 30px 20px;
  position: relative;

  @media (min-width: ${theme.breakpoints.md}) {
    height: auto;
    max-height: none;
    min-height: auto;
    overflow-x: auto;
    overflow-y: hidden;
    padding: 50px;
  }

  h1.title {
    margin-top: 0;
  }
`
