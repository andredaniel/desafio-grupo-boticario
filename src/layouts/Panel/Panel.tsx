import React from 'react'
import { AppBar } from '../../containers/AppBar'
import { NavBar } from '../../containers/NavBar'
import { AppContent, PanelWrapper } from './Panel.styles'

const Panel: React.FC = ({ children }): JSX.Element => {
  return (
    <PanelWrapper>
      <NavBar />
      <AppBar />
      <AppContent>{children}</AppContent>
    </PanelWrapper>
  )
}

export { Panel }
