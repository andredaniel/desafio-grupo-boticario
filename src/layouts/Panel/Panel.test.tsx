import { render } from '@testing-library/react'
import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { UserProvider } from '../../contexts/user.context'
import { Panel } from './Panel'

describe('<Panel />', () => {
  it('should render Panel component', () => {
    const { container } = render(
      <UserProvider>
        <Router>
          <Panel />
        </Router>
      </UserProvider>
    )
    expect(container).toMatchSnapshot()
  })
})
