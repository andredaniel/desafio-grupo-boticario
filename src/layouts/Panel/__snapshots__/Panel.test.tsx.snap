// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Panel /> should render Panel component 1`] = `
<div>
  <div
    class="sc-jrAGrp iRohlz"
  >
    <div
      class="sc-pFZIQ fgkQxn"
      y="0"
    >
      <div
        class="brand-wrapper"
      >
        <svg>
          logo-black.svg
        </svg>
        <span>
          Cashback
        </span>
      </div>
      <div
        class="user-info"
      >
        <img
          class="sc-hKgILt dXcvOA"
          src=""
        />
        <span
          class="welcome-message"
        >
          Ol√°, 
          !
        </span>
      </div>
    </div>
    <div
      class="sc-bdfBwQ cTIOBo"
    >
      <div
        class="sc-gsTCUz ifxEXl"
      >
        <div
          class="sc-dlfnbm goBrec"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 9l9-7 9 7v11a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
              />
              <polyline
                points="9 22 9 12 15 12 15 22"
              />
            </svg>
            <span>
              Dashboard
            </span>
            <canvas
              height="0"
              style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
              width="0"
            />
          </a>
        </div>
        <div
          class="sc-dlfnbm goBrec"
        >
          <a
            href="/orders/create"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="18"
                rx="2"
                ry="2"
                width="18"
                x="3"
                y="3"
              />
              <line
                x1="12"
                x2="12"
                y1="8"
                y2="16"
              />
              <line
                x1="8"
                x2="16"
                y1="12"
                y2="12"
              />
            </svg>
            <span>
              Nova compra
            </span>
            <canvas
              height="0"
              style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
              width="0"
            />
          </a>
        </div>
        <div
          class="sc-dlfnbm goBrec"
        >
          <a
            href="/orders"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
              />
              <line
                x1="3"
                x2="21"
                y1="6"
                y2="6"
              />
              <path
                d="M16 10a4 4 0 0 1-8 0"
              />
            </svg>
            <span>
              Minhas compras
            </span>
            <canvas
              height="0"
              style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
              width="0"
            />
          </a>
        </div>
        <div
          class="sc-dlfnbm goBrec"
        >
          <a
            href="/logout"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
              />
              <polyline
                points="16 17 21 12 16 7"
              />
              <line
                x1="21"
                x2="9"
                y1="12"
                y2="12"
              />
            </svg>
            <span>
              Sair
            </span>
            <canvas
              height="0"
              style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
              width="0"
            />
          </a>
        </div>
      </div>
    </div>
    <div
      class="sc-kEjbxe hCePyy"
    />
  </div>
</div>
`;
